### PR TITLE
#164069178 Users Signup

### PR DESCRIPTION
**Pull request by joelethan**
### What does this PR do?
- This PR creates a signup page used by users to register with Authors Haven 

### Description of Task to be completed?
- Provide the user with a signup form
- Provide descriptive error messages
- Reroute the user to the login page on email verification

### How should this be manually tested?
- Clone the [repo](https://github.com/andela/ah-aquaman-frontend) and cd into ah-aquaman-frontend
- checkout the branch `ft-users-should-signup-164069178`
- Run the commands below to install packages and run the server
          - npm install
          - npm start
- Navigate to `/signup` 
- Provide `username`, `email` and `password` to register
- This feature is hosted [here](https://ah-aquaman-frontend-stag-pr-12.herokuapp.com/signup)

### What are the relevant pivotal tracker stories?

- [#164069178](https://www.pivotaltracker.com/story/show/164069178)

### Screenshots
Screenshot 2019-04-11 at 07 57 50

### CODE REVIEW

File: src/__tests__/views/signupView/__test__/signupView.test.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+};
+const token = generateToken(user);
+
+describe("SignupView", () => {
+  it("should render without crashing", () => {
+    const props = {};
+    const instance = new SignupView(props);
+    const wrapper = shallow(instance.render());
+    expect(wrapper).toMatchSnapshot();
+  });
```
Comment (wasibani-roy) [inline src/__tests__/views/signupView/__test__/signupView.test.js:35]: Please add a new line so as to improve code readability

File: src/__tests__/views/signupView/__test__/signupView.test.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+import { shallow, mount } from "enzyme";
+import { Provider } from "react-redux";
+import configureStore from "redux-mock-store";
+import { SignupView } from "../../../../views/signupView";
+
+const jwt = require("jsonwebtoken");
+
+const props = {
+  signupAction: jest.fn(),
+  toast: jest.fn(),
```
Comment (wasibani-roy) [inline src/__tests__/views/signupView/__test__/signupView.test.js]: Do we really have to define toast here as a prop seeing as its, not in the props on the view

File: src/components/signupForm/index.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+import PropTypes from "prop-types";
+import InputField from "../../commons/inputField";
+import Label from "../../commons/label";
+
+const Signup = (props) => {
+  const {
+    onSubmit, errors, onClick, isProcessing, 
+  } = props;
+
+  const title = "Author's Haven";
```
Comment (wasibani-roy) [inline src/components/signupForm/index.js:11]: Please add a line here

**Review by wasibani-roy — CHANGES_REQUESTED** [review]
Good work @joelethan just a few changes needed

**Review by CryceTruly — CHANGES_REQUESTED** [review]
Great work @joelethan,  Lets work on responsiveness on small screen devices.

**Review by CryceTruly — COMMENTED** [review]
Great work @joelethan, the feature is almost flawless, but I think we should be taken to the login page after the successful signup, not the homepage. What do you think?

**Review by wasibani-roy — APPROVED** [review]

**Comment by joelethan** [comment]
@CryceTruly It is already redirecting to the login page.

**Review by Peace-Apple — APPROVED** [review]
Good work @joelethan thanks for implementing the feedback

**Review by CryceTruly — APPROVED** [review]

File: src/__tests__/actions/signupActions/__test__/signupActions.test.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import configureMockStore from "redux-mock-store";
+import thunk from "redux-thunk";
+import fetchMock from "fetch-mock";
+import signupAction from "../../../../actions/signupAction";
+import { REGISTER_SUCCESS, REGISTER_FAILURE } from "../../../../actions/types";
+
+
```
Comment (AmosWels) [inline src/__tests__/actions/signupActions/__test__/signupActions.test.js]: remove all unnecessary lines in the code

File: src/__tests__/actions/signupActions/__test__/signupActions.test.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    type: REGISTER_FAILURE,
+    payload: {},
+  }];
+
+  const invalidData = {
+    email: "",
+    password: "",
+  };
+
+  const store = mockStore();
```
Comment (AmosWels) [inline src/__tests__/actions/signupActions/__test__/signupActions.test.js]: You can create one const for `store` in this file in the describe

**Review by AmosWels — CHANGES_REQUESTED** [review]
Goodwork @joelethan , resolve the eslint issues in your PR

**Review by AmosWels — COMMENTED** [review]
Please move the `signupReducer.test.js` file to the reducers folder in your tests

File: src/__tests__/actions/signupActions/__test__/signupActions.test.js (hunk 5 above)
Comment (joelethan) [reply src/__tests__/actions/signupActions/__test__/signupActions.test.js]: I think it is better to have different stores so as to isolate the data between different tests.

**Review by AmosWels — APPROVED** [review]

**Pull request merged**